### PR TITLE
Jenkins single node deploy and destroy 

### PR DESCRIPTION
**Pull request by sradhanand-pati**
Signed-off-by: sradhanandpati <sradhanand.pati@seagate.com>

### CODE REVIEW

File: devops/ci/single_node/destroy/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env groovy
+
+node {
+    properties([
+        buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '', daysToKeepStr: '5', numToKeepStr: '5')),
+        parameters([
+            string(defaultValue: '', description: 'FQDN of VM to deploy on.', name: 'HOST_NAME', trim: true),
```
Comment (ypise) [inline devops/ci/single_node/destroy/Jenkinsfile:7]: `Deploy on` in destroy?

File: devops/ci/single_node/destroy/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                echo "Sucessfully stopped salt and gluster services"
+            }
+            stage("Uninstall rpms") {
+                sshCommand remote: remote, command: """
+                    yum erase -y cortx-prvsnr cortx-prvsnr-cli      # Cortx Provisioner packages
+                    yum erase -y gluster-fuse gluster-server        # Gluster FS packages
+                    yum erase -y salt-minion salt-master salt-api   # Salt packages
+                    yum erase -y python36-m2crypto                  # Salt dependency
+                    yum erase -y python36-cortx-prvsnr              # Cortx Provisioner API packages
+                    yum erase -y *cortx*
```
Comment (ypise) [inline devops/ci/single_node/destroy/Jenkinsfile:47]: Add comment:
Add brute clean for cortx:
```
# Brute clean for any cortx rpm packages
yum erase -y *cortx*
```

File: devops/ci/single_node/destroy/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                    yum autoremove -y
```
Comment (ypise) [inline devops/ci/single_node/destroy/Jenkinsfile:48]: Add comment:
```
# Re-condition yum db
yum autoremove -y
yum clean all
rm -rf /var/cache/yum
```

**Review by ypise — CHANGES_REQUESTED** [review]

File: devops/ci/single_node/deploy/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                python_deps/
+                README.txt
+                RELEASE.INFO
+                THIRD_PARTY_RELEASE.INFO''', name: 'BUILD_URL', trim: true),
+            password(description: 'root user password for the target node.', name: 'PASSWORD')
+        ])
+    ])
+
+    def remote = [:]
+    withEnv(["CORTX_RELEASE_REPO=${BUILD_URL}","SSHPASS=${PASSWORD}"]) {
```
Comment (andkononykhin2) [inline devops/ci/single_node/deploy/Jenkinsfile:21]: makes sense to think about `withCredentials`

**Review by andkononykhin2 — COMMENTED** [review]
@sradhanand-pati  Looks accurate. Thanks. I believe we need a follow-up task to simplify destroy logic later basing API for that (#851)

CC @ypise

File: devops/ci/single_node/deploy/Jenkinsfile (hunk 4 above)
Comment (ypise) [reply devops/ci/single_node/deploy/Jenkinsfile:21]: withCredentials has a bug, moreover PASSWORD is used through the `sshCommand`  plugin.
We might want to mask password though.

File: devops/ci/single_node/destroy/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    
+    def remote = [:]
+    withEnv(["SSHPASS=${PASSWORD}"]) {
+        ansiColor('xterm') {
+            remote.name = "srvnode-1"
+            remote.host = HOST_NAME
+            remote.user = 'root'
+            remote.password = PASSWORD
+            remote.allowAnyHosts = true
+            stage("SSH Connectivity Check") {
```
Comment (ypise) [inline devops/ci/single_node/destroy/Jenkinsfile:20]: Check for empty password just like in deploy file:
```
if (PASSWORD.isEmpty()) {
    error "Target VM password cannot be empty."
}
```

**Review by ypise — CHANGES_REQUESTED** [review]

File: devops/ci/single_node/destroy/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                echo "Successfully uninstalled rpms"
+            }
+            stage("Cleanup bricks and other directories") {
+                sshCommand remote: remote, command: """
+                    # Cortx software dirs
+                    rm -rf /opt/seagate/cortx
+                    rm -rf /opt/seagate/cortx_configs
+                    rm -rf /opt/seagate
+                    # Bricks cleanup
+                    test -e /var/lib/seagate && rm -rf /var/lib/seagate
```
Comment (ypise) [inline devops/ci/single_node/destroy/Jenkinsfile:70]: If test fails Jenkins job exits.
Append  `|| true` to ensure job continuity for all statements with `test -e` checks.

**Review by ypise — CHANGES_REQUESTED** [review]

**Review by ypise — APPROVED** [review]

File: devops/ci/destroy/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env groovy
+
```
Comment (ypise) [inline devops/ci/destroy/Jenkinsfile:2]: Rename this file to:
 devops/ci/single_node_deploy/Jenkinsfile_Deploy

Also, add copyright headers.

File: devops/ci/single_node_deploy/Jenkinsfile_single_node_deploy [hunk 8]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env groovy
+
```
Comment (ypise) [inline devops/ci/single_node_deploy/Jenkinsfile_single_node_deploy:2]: Rename this file to:
 devops/ci/single_node_deploy/Jenkinsfile_Destroy

Also, add copyright headers.

**Review by ypise — CHANGES_REQUESTED** [review]

**Review by ypise — APPROVED** [review]

**Pull request merged**